salt/utils/event.py

```diff
@@ -121,6 +121,14 @@ def get_event(
                            set_event_handler() API. Otherwise, operation
                            will be synchronous.
     """
+    if "transport" in kwargs:
+        salt.utils.versions.warn_until(
+            "3006",
+            "The 'transport' kwarg has been deprecated and its "
```

Comment:
When deprecating functionality you should not also remove it at the same time. First we need to add this deprecation notice that it will be removed and then in the 3006 release then we remove the kwarg and remove the deprecation warning.

Reply:
Sure, yeah, that makes sense to me.